staging/src/k8s.io/cluster-bootstrap/token/util/helpers.go

```diff
@@ -131,3 +134,80 @@ func ValidateUsages(usages []string) error {
 	}
 	return nil
 }
+
+// GetSecretString returns the string value for the given key in the specified Secret
+// If there is an error or if the key doesn't exist, an empty string is returned.
+func GetSecretString(secret *v1.Secret, key string) string {
+	if secret.Data == nil {
+		return ""
+	}
+	if val, ok := secret.Data[key]; ok {
+		return string(val)
+	}
+	return ""
+}
+
+// IsSecretExpired will identify whether the secret expires
+func IsSecretExpired(secret *v1.Secret) bool {
+	expiration := GetSecretString(secret, api.BootstrapTokenExpirationKey)
+	if len(expiration) > 0 {
+		expTime, err2 := time.Parse(time.RFC3339, expiration)
+		if err2 != nil {
+			klog.V(3).Infof("Unparseable expiration time (%s) in %s/%s Secret: %v. Treating as expired.",
+				expiration, secret.Namespace, secret.Name, err2)
+			return true
+		}
+		if time.Now().After(expTime) {
+			klog.V(3).Infof("Expired bootstrap token in %s/%s Secret: %v",
+				secret.Namespace, secret.Name, expiration)
+			return true
+		}
+	}
+	return false
+}
+
+// ParseToken tries and parse a valid token from a string.
+// A token ID and token secret are returned in case of success, an error otherwise.
+func ParseToken(s string) (string, string, error) {
+	split := BootstrapTokenRegexp.FindStringSubmatch(s)
+	if len(split) != 3 {
+		return "", "", fmt.Errorf("token [%q] was not of form [%q]", s, api.BootstrapTokenPattern)
+	}
+	return split[1], split[2], nil
+}
+
+// ParseSecretName parses the name of the secret to extract the secret ID.
+func ParseSecretName(name string) (secretID string, ok bool) {
+	namePattern := `^` + regexp.QuoteMeta(api.BootstrapTokenSecretPrefix) + `([a-z0-9]{6})$`
+	nameRegExp := regexp.MustCompile(namePattern)
```

Comment:
I would not recommend using MustCompile on a constructed regexp, even if this one looks pretty safe.